.github/workflows/on-change-check-frontend.yml

```diff
@@ -0,0 +1,34 @@
+name: 'On change: run check-frontend'
```

Comment:
Should the name of the workflow also convey when the workflow runs? If we enable it to run on dispatch, the name will be misleading. We should just name them after what they do: "Frontend checks", etc.

Reply:
I was trying to solve for the awkward interactions between how workflows get triggered without duplicating the workflow logic. 

For example, the pytest workflow should be run:
- on dispatch, always
- on call, always
- on code change, if the code change touches the app code

The changed files action complicates things by failing (not gracefully!) when used in a dispatched workflow because the last commit and target git ref are the same. It sees no changes and fails. 

First I tried using a series of "force" or "skip" inputs but this got really convoluted.

Then I learned about `github.event_name` (to only run on call/dispatch). But that's also messy, adding a lot of ifs. Maybe it's less messy than what I ended up with...

...which is splitting the workflow "business logic" from the "should I run this" logic. Unfortunately this is hard to cleanly represent and I ended up with two workflows for pytest:

1. a workflow that only runs on call or dispatch and always runs pytest
2. a workflow that runs said workflow on code changes

So there's a "Check: pytest" that only runs on call/dispatch and "On change: pytest" that runs on code changes.

(This all applies to the frontend and python code checks too)

What would be the conventional way to handle this situation? This is my first rodeo 😅

Reply:
Oh - it would be much easier if pytest could be a composite action - but it's a job with a matrix, so it's hard to compose...

Reply:
thanks for explaining - the naming decision makes way more sense now.

> What would be the conventional way to handle this situation?

there isn't one :smiling_face_with_tear:  Every CI system likes its own way of doing things. With GHA, your experiences of `this got really convoluted.` > `that's also messy, adding a lot of ifs` > `this is hard to cleanly represent` is par for the course :)

I think there should be a way to further extract `pytest` jobs from workflows into actions the way you did with dependencies.... Need to process.

Reply:
Let's keep this in mind for future enhancements